### ISSUE TITLE
npm should use Latest as Wanted for globally installed packages

### ISSUE THREAD

**Issue by cloudlena** [npm/npm#19888]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

When running `npm outdated -g`, npm sometimes lists newer versions as the `Wanted` version than the `Latest` version. I think, for globally installed packages, `Wanted` should always be `Latest`. Otherwise, the `Wanted` field doesn't make too much sense for globally installed ones.

This would also result in `npm update -g` not updating to versions which are newer than the latest.

#### How can the CLI team reproduce the problem?

Only works at the moment because somehow npm 5.7.0 is not flagged as `latest` yet (despite it having been released):

```shell
$ npm install -g npm@latest
$ npm outdated -g
Package  Current  Wanted  Latest  Location
npm      5.6.0    5.7.0   5.6.0
$ npm update -g # Installs npm 5.7.0 instead of the expected 5.6.0
```

### supporting information:

 - `npm -v` prints: `5.6.0`
 - `node -v` prints: `v9.5.0`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: macOS 10.13.3
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by pakastin** [comment]
https://docs.npmjs.com/cli/outdated
> wanted is the maximum version of the package that satisfies the semver range specified in package.json. If there's no available semver range (i.e. you're running npm outdated **--global**, or the package isn't included in package.json), then wanted shows the **currently-installed version**
🤔

**Comment by cloudlena** [comment]
Thanks, @pakastin. As you can see in my example above, that's not what it does. First I'm running `npm install -g npm@latest`. At this point, 5.6.0 is `latest` so it gets installed. Running `npm outdated -g` right after still returns 5.7.0 as `Wanted` (despite 5.6.0 being installed).

I stumbled upon this issue when running `npm update -g` this morning and npm 5.7.0 was installed which I didn't expect because 5.6.0 is still `latest`.

**Comment by pakastin** [comment]
Yeah, that's not how it should work. There's definitely a bug and can confirm that it occurs with others as well. However for some reason I can't get it reproduced.

**Comment by petetnt** [comment]
Can confirm this behaviour with npm@5.6.0

![image](https://user-images.githubusercontent.com/7641760/36541168-b495a97a-17e5-11e8-9cb0-e520f49be837.png)

Related to this too: https://github.com/npm/npm/issues/19883

**Comment by dougmoscrop** [comment]
[This](https://github.com/npm/npm/blob/latest/lib/outdated.js#L384) is at least one of the places where this happens, because `r` will be something like `^5.6.0`, and so `maxSatisfying` is, of course going to return something like `5.7.0` as `versions` is every version without any kind of dist-tag context. You could maybe .`filter(v => semver.lte(v, l)` but this isn't the only code path..

I think doing it this way, with effectively a filter on semver values that have lost their release tags is not ideal; I say this because teams might manage multiple release-stages (think 2.0 and 1.0) and this filter/max approach basically expects that you only publish versions that are larger. [this comment](https://github.com/npm/npm/blob/latest/lib/outdated.js#L226) seems poignant now -- update should probably be changed to just install whatever a given `dist-tag` (defaulting to `latest`) is currently pointed at, as long as it satisfies wanted. This would fit with the clause of "it will be downgraded" (i.e. if you installed a prerelease, but then do update @ latest -- you would do update `@next` for example to preserve your dist-tag branch) in the docs.

**Comment by joeycozza** [comment]
Looks like they have a PR for this now. https://github.com/npm/npm/pull/19892
